tests/parsenum/main.c

```diff
@@ -0,0 +1,97 @@
+#include <stdint.h>
+#include <stdio.h>
+
+#include "parsenum.h"
+#include "warnp.h"
+
+#define PARSENUM_CMP(x, y)						\
+	((isnan((double)x) && isnan((double)y)) ||			\
+	    (isinf((double)x) == isinf((double)y) && isinf((double)x)) || \
```

Comment:
Growl.  glibc says:
> isinf(x)      returns 1 if x is positive infinity, and -1 if x is negative infinity.

but I just checked in POSIX and it makes no such guarantees; merely
> The isinf() macro shall return a non-zero value if and only if its argument has an infinite value.